gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagProcessingEngine.java

```diff
@@ -149,6 +158,16 @@ public void run() {
           dagTask.conclude();
           log.info(dagProc.contextualizeStatus("concluded dagTask"));
         } catch (Exception e) {
+          log.error("DagProcEngineThread: " + dagProc.contextualizeStatus("error"), e);
+          if (!DagProcessingEngine.isTransientException(e)) {
+            DagActionStore.DagAction dagAction = dagTask.getDagAction();
+            if (dagAction != null) {
+              log.warn(dagProc.contextualizeStatus("ignoring non-transient exception by concluding so no retries"));
+            }
+            dagManagementStateStore.getDagManagerMetrics().dagProcessingNonRetryableExceptionMeter.mark();
+            dagTask.conclude();
+          }
+          // TODO add the else block for transient exceptions and add conclude task only if retry limit is not breached
           log.error("DagProcEngineThread: " + dagProc.contextualizeStatus("error"), e);
```

Comment:
would be clearer to reorder and present the exception w/ stack trace first, before describing how the non-transient subset are getting special handling

Reply:
actually seems you did reorder... but this is actually a dup.  keep only the first and also move this up w/ it, before the special handling:
```
 dagManagementStateStore.getDagManagerMetrics().dagProcessingExceptionMeter.mark();
```

Reply:
Sure updated